File: mario.c
#include <cs50.h>
#include <stdio.h>

void printSpaces(int s);
void printHashes(int h);

int main(void)
{
    int height;
    int spaces;
    int hashes;
    int row;
    do
    {
        height = get_int("How tall would you like the pyramid? ");
    }
    while (height < 1 || height > 8);
    row = 1;
    hashes = 1;
    spaces = height - row;
    while (row <= height)
    {
        printSpaces(spaces);
        printHashes(hashes);
        printf("  ");
        printHashes(hashes);
        printf("\n");
        row++;
        hashes++;
        spaces--;
    }
}

// Create a function to print spaces a given number of times
void printSpaces(int s)
{
    for (int i = 0; i < s; i++)
    {
        printf(" ");
    }
}

// Create a function that prints hashes a given number of times
void printHashes(int h)
{
    for (int j = 0; j < h; j++)
    {
        printf("#");
    }
}

File: mario.py
from cs50 import get_int


def main():
    while True:
        height = get_int("Height: ")
        if height >= 1 and height <= 8:
            break
    row = 1
    hashes = 1
    spaces = height - row
    while hashes <= height:
        print_spaces(spaces)
        print_hashes(hashes)
        print("  ", end="")
        print_hashes(hashes)
        print()
        row += 1
        hashes += 1
        spaces -= 1


def print_spaces(n):
    for i in range(n):
        print(" ", end="")


def print_hashes(n):
    for i in range(n):
        print("#", end="")


main()
